token/js/client/token.js

```diff
@@ -458,6 +482,101 @@ export class Token {
     return newAccount.publicKey;
   }
 
+  /**
+   * Create and initialize the associated account.
+   *
+   * This account may then be used as a `transfer()` or `approve()` destination
+   *
+   * @param owner User account that will own the new account
+   * @return Public key of the new associated account
+   */
+  async createAssociatedTokenAccount(owner: PublicKey): Promise<PublicKey> {
+    const associatedAddress = await Token.getAssociatedTokenAddress(
+      this.associatedProgramId,
+      this.programId,
+      this.publicKey,
+      owner,
+    );
+
+    return this.createAssociatedTokenAccountInternal(owner, associatedAddress);
+  }
+
+  async createAssociatedTokenAccountInternal(
+    owner: PublicKey,
+    associatedAddress: PublicKey,
+  ): Promise<PublicKey> {
+    await sendAndConfirmTransaction(
+      'CreateAssociatedTokenAccount',
+      this.connection,
+      new Transaction().add(
+        Token.createAssociatedTokenAccountInstruction(
+          this.associatedProgramId,
+          this.programId,
+          this.publicKey,
+          associatedAddress,
+          owner,
+          this.payer.publicKey,
+        ),
+      ),
+      this.payer,
+    );
+
+    return associatedAddress;
+  }
+
+  /**
+   * Retrieve the associated account or create one if not found.
+   *
+   * This account may then be used as a `transfer()` or `approve()` destination
+   *
+   * @param owner User account that will own the new account
+   * @return The new associated account
+   */
+  async getOrCreateAssociatedAccountInfo(
+    owner: PublicKey,
+  ): Promise<AccountInfo> {
+    const associatedAddress = await Token.getAssociatedTokenAddress(
+      this.associatedProgramId,
+      this.programId,
+      this.publicKey,
+      owner,
+    );
+
+    // This is the optimum logic, considering TX fee, client-side computation,
+    // RPC roundtrips and guaranteed idempotent.
+    // Sadly we can't do this atomically;
+    try {
+      return await this.getAccountInfo(associatedAddress);
+    } catch (err) {
+      // INVALID_ACCOUNT_OWNER can be possible if the associatedAddress has
+      // already been received some lamports (= became system accounts).
+      // Assuming program derived addressing is safe, this is the only case
+      // for the INVALID_ACCOUNT_OWNER in this code-path
+      if (
+        err.message === FAILED_TO_FIND_ACCOUNT ||
```

Comment:
(small fyi: btw because we take no user-provided callbacks, this is safe; otherwise, we must guard against `throw 3` or `throw undefined`, iirc; nasty JavaScript... lol)